core/src/apresolve.rs

```diff
@@ -114,9 +140,9 @@ impl ApResolver {
                 // take the first position instead of the last with `pop`, because Spotify returns
                 // access points with ports 4070, 443 and 80 in order of preference from highest
                 // to lowest.
-                "accesspoint" => inner.data.accesspoint.remove(0),
-                "dealer" => inner.data.dealer.remove(0),
-                "spclient" => inner.data.spclient.remove(0),
+                "accesspoint" => inner.data.accesspoint.first(),
```

Comment:
The intention here is that the access points are drained, for a future implementation of reconnection logic. If we successfully connect to an AP, but it fails later on, we want to reconnect taking the next in the list.

So maybe not do `first` here but `next` -- if so are we sure that the cursor remains there for the above to work?

Reply:
Ah I see. I was thinking about what would happen on multiple calls to the function. The main `"accesspoint"` is called inside the `Session::connect` function, is a safe assumption to make that `connect` is only called once? If somebody would want to reconnect, it would use another AP even if the first one worked.

In any case, since it is actually intended to be drained, I would switch to a `VecDeque` instead of a `Vec`. That allows for ergonomic draining from the front.

Reply:
It’s only once now, but we intend to make it so we can reconnect. This is a long-standing issue that we’re laying the pavement for here.